pinner/pinning_op/pinning_op.go

```diff
@@ -36,24 +36,42 @@ type PinningOperation struct {
 
 	SkipLimiter bool
 
-	Lk sync.Mutex
+	lk sync.Mutex
 
 	MakeDeal bool
 }
 
 func (po *PinningOperation) Fail(err error) {
-	po.Lk.Lock()
+	po.lk.Lock()
+	defer po.lk.Unlock()
+
 	po.FetchErr = err
 	po.EndTime = time.Now()
 	po.Status = types.PinningStatusFailed
 	po.LastUpdate = time.Now()
```

Comment:
nit: can we update the status before these timestamps and use the same timestamp for EndTime and LastUpdate?